roms_tools/setup/datasets.py

```diff
@@ -160,12 +154,18 @@ def load_data(self) -> xr.Dataset:
 
         # Handle the case when filename is a list
         elif isinstance(filename_str, list):
-            if not self.use_dask:
-                raise ValueError(
-                    "A list of files is provided, but use_dask=False. "
-                    "Multiple files can only be loaded when use_dask=True."
-                )
-            matching_files = filename_str
+            # Check if any element in the list contains wildcard characters
+            if any(re.search(r"[\*\?\[\]]", f) for f in filename_str):
```

Comment:
You do the same regex search more than once so you might be able to cache it using a different function from the regex std lib.

Reply:
Good point. Done in 22e7b7c.